katharsis-core/src/main/java/io/katharsis/queryspec/InMemoryEvaluator.java

```diff
@@ -97,11 +97,15 @@ private static boolean matchesPrimitiveOperator(Object object, FilterSpec filter
 		Object value = PropertyUtils.getProperty(object, filterSpec.getAttributePath());
 		FilterOperator operator = filterSpec.getOperator();
 		Object filterValue = filterSpec.getValue();
-		if (value instanceof Collection) {
-			return matchesAny((Collection<?>) value, operator, filterValue);
-		}
-		else {
-			return operator.matches(value, filterValue);
+		if (filterValue != null ||  (filterValue == null && (operator.equals(FilterOperator.EQ) || operator.equals(FilterOperator.NEQ)))) {
```

Comment:
Definitely better without all the branching. your `matchesAny` function is `((Collection)value).stream().anyMatch(v -> operator.matches(v,filterValue))`

Is there a reason you're not using a lambda and just getting rid of that function?